recipe/patches-grpcio/0001-fix-win-setup.patch

```diff
@@ -0,0 +1,48 @@
+--- setup.py	2021-07-20 19:26:11.000000000 -0400
++++ setup.py	2021-07-24 15:17:29.916938943 -0400
+@@ -23,7 +23,6 @@
+ UnixCCompiler.src_extensions.append('.S')
+ del UnixCCompiler
+
+-from distutils import cygwinccompiler
+ from distutils import extension as _extension
+ from distutils import util
+ import os
+@@ -197,6 +196,9 @@
+
+
+ def check_linker_need_libatomic():
++    if sys.platform == 'win32':
++        return False
++
+     """Test if linker on system needs libatomic."""
+     code_test = (b'#include <atomic>\n' +
+                  b'int main() { return std::atomic<int64_t>{}; }')
+@@ -229,7 +231,7 @@
+ EXTRA_ENV_COMPILE_ARGS = os.environ.get('GRPC_PYTHON_CFLAGS', None)
+ EXTRA_ENV_LINK_ARGS = os.environ.get('GRPC_PYTHON_LDFLAGS', None)
+ if EXTRA_ENV_COMPILE_ARGS is None:
+-    EXTRA_ENV_COMPILE_ARGS = ' -std=c++11'
++    EXTRA_ENV_COMPILE_ARGS = ' '
+     if 'win32' in sys.platform:
+         if sys.version_info < (3, 5):
+             EXTRA_ENV_COMPILE_ARGS += ' -D_hypot=hypot'
+@@ -241,10 +243,6 @@
+                 EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime32 -D_timeb=__timeb32 -D_ftime_s=_ftime32_s'
+             else:
+                 EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime64 -D_timeb=__timeb64'
+-        else:
+-            # We need to statically link the C++ Runtime, only the C runtime is
+-            # available dynamically
+-            EXTRA_ENV_COMPILE_ARGS += ' /MT'
+     elif "linux" in sys.platform:
+         EXTRA_ENV_COMPILE_ARGS += ' -std=gnu99 -fvisibility=hidden -fno-wrapv -fno-exceptions'
+     elif "darwin" in sys.platform:
+@@ -257,6 +255,7 @@
+         if check_linker_need_libatomic():
+             EXTRA_ENV_LINK_ARGS += ' -latomic'
+     elif "win32" in sys.platform and sys.version_info < (3, 5):
```

Comment:
Ha, is grpcio compatible with Python < 3.5?

Reply:
according to their classifiers it isn't anymore, but this context still exists on the current head.